x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/settings_flyout/alert_selection/use_data_view/index.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DataView, DataViewSpec } from '@kbn/data-views-plugin/common';
+import { useEffect, useState } from 'react';
+
+import { useKibana } from '../../../../../common/lib/kibana';
+
+export const useDataView = ({
+  dataViewSpec,
+  loading,
+}: {
+  dataViewSpec: DataViewSpec | undefined;
```

Comment:
In `8.x`, when the `AlertSelectionQuery` component in `x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/settings_flyout/alert_selection/alert_selection_query/index.tsx` obtains a `sourcererDataView` via `useSourcererDataView` in the following code:

```typescript
  const { sourcererDataView, loading: isLoadingIndexPattern } = useSourcererDataView(
    SourcererScopeName.detections
  );
```

the `sourcererDataView` is typed as:

```typescript
const sourcererDataView: DataViewSpec | undefined
```

The `dataViewSpec` parameter of the `useDataView` hook was updated (in this 8.x branch) to reflect the fact that `sourcererDataView` may be undefined:

```typescript
dataViewSpec: DataViewSpec | undefined;
```